### DISCUSSION TITLE
data.js redundant properties

### DISCUSSION THREAD

**Discussion by f0nar** [4ian/GDevelop#6562] [Technical questions about the codebase]
Hi!
I created custom extension that includes functions with inline code and custom object (object uses functions). I dont want to share it here as it is super specific testing example for custom needs and I am sure that it will confuse you even more. I think extension structure is the key
<img width="1178" alt="Screenshot 2024-05-13 at 14 07 27" src="https://github.com/4ian/GDevelop/assets/48058174/d01f71e5-fd33-42f9-8187-9663a8353314">

Then I exported my game as HTML5 game export and found an interesting thing
All of my inline code which is more then 2.4k lines was wrapped and placed in `gdjs-evtsext__...smth...-func.js` file which is OK. But then I realized that it is duplicated in `data.js` file in `projectData.eventsFunctionsExtensions.eventsFunctions` property. I checked GDevelop repo and found that `eventsFunctionsExtensions` property is typed as `EventsFunctionsExtensionData[]` where 
```
declare interface EventsFunctionsExtensionData {
  name: string;
  eventsBasedObjects: EventsBasedObjectData[];
}

declare interface EventsBasedObjectData {
  name: string;
  objects: Array<ObjectData & any>;
}
``` 
so this duplicated code is not used in runtime in any way as I understand. So why is it there as well as a lot of other properties which are not specified in `EventsFunctionsExtensionData` type? 
<img width="621" alt="Screenshot 2024-05-13 at 14 25 35" src="https://github.com/4ian/GDevelop/assets/48058174/2a673a4b-80d8-4722-9d00-729b5577746d">

In addition I see that there is no such duplications for my other extension that define functions with inline code (but not custom object). Is this due to usage in custom object?

**Comment by D8H** [comment]
Thank you for reporting this issue, it will be solved by this PR:
- https://github.com/4ian/GDevelop/pull/6459

**Reply by f0nar** [reply]
sounds great. will it solve issue with all properties that are unused in runtime?

**Reply by D8H** [reply]
It removes `eventFunctions` content.

**Reply by f0nar** [reply]
what about other properties like author, category and so on? are they really needed in runtime?

**Reply by 4ian** [reply]
Probably a good idea to have a method `StripForExport` for these data. 
`StripProjectForExport` can call them on each extension and other classes (I'm not talking about a generic solution, just a bunch of loops and method calls).

cc @D8H let's do this?

**Reply by f0nar** [reply]
that sounds great
